api/v1alpha1/databaseclusterrestore_types.go

```diff
@@ -37,6 +40,53 @@ type DatabaseClusterRestoreStatus struct {
 	Message     string       `json:"message,omitempty"`
 }
 
+// PITR represents a specification to configure point in time recovery for a database backup/restore.
+type PITR struct {
+	// Type is the type of recovery. pxc: full list; psmdb, pg: date,latest
+	// +kubebuilder:validation:Enum:=date;latest;transaction;skip
+	// +kubebuilder:default:=date
+	Type string `json:"type,omitempty"`
+	// Date is the date to recover to
+	Date *RestoreDate `json:"date,omitempty"`
+	// GTID is the transaction ID to restore to (pxc only)
+	GTID string `json:"gtid,omitempty"`
```

Comment:
Transaction level selection isn't part of the scope for the first iteration of PITR we'll only have the date and time option. Maybe we should leave this out of the picture for now.

Reply:
Sure, I love deleting things :) Done